src/main/scala/pt/tecnico/dsi/neutron/models/Network.scala

```diff
@@ -1,58 +1,62 @@
 package pt.tecnico.dsi.neutron.models
 
-import java.time.LocalDateTime
-
 import io.circe.derivation.{deriveDecoder, deriveEncoder, renaming}
-import io.circe.{Decoder, Encoder, HCursor}
+import io.circe.{Decoder, Encoder}
 
 object Network {
 
-  // TODO: Rethink this
   object Read {
-    implicit val decoder: Decoder[Read] = (c: HCursor) => {
-      for {
-        providerNetworkType <- c.downField("provider:network_type").as[String]
-        providerPhysicalNetwork <- c.downField("provider:physical_network").as[String]
-        providerSegmentationId <- c.downField("provider:segmentation_id").as[Integer]
-        routerExternal <- c.downField("router:external").as[Boolean]
-        network <- c.as[Read](deriveDecoder(renaming.snakeCase))
-      } yield network.copy(
-        providerNetworkType = providerNetworkType,
-        providerPhysicalNetwork = providerPhysicalNetwork,
-        providerSegmentationId = providerSegmentationId,
-        routerExternal = routerExternal
-      )
+
+    def decoderAfterRename[T](m: Map[String, String], d: Decoder[T]): Decoder[T] = {
+      d.prepare {
+        _.withFocus {
+          _.mapObject { x =>
+            m.foldLeft(x) { (a, b) =>
+              val value = x(b._1)
+              a.add(b._2, value.get).remove(b._1)
```

Comment:
Codacy found an issue: [Found unnamed result or parameter](https://app.codacy.com/gh/ist-dsi/scala-neutron-client/pullRequest?prid=5827787)

Reply:
Click the link then click the error inside Codacy, there is a very good explanation. In this case the solution will be something like:

```scala
m.foldLeft(x) { (accumulator, (name, correctName)) =>
  val value = x(name)
  a.add(correctName, value.get).remove(name)
}
```

However that get will be troublesome. I recommend this:
```scala
  def decoderAfterRename[T](renames: Map[String, String], d: Decoder[T]): Decoder[T] = d.prepare(_.withFocus(_.mapObject{ obj =>
    val newMap = obj.toMap.map { case (key, value) =>
      renames.getOrElse(key, key) -> value
    }
    JsonObject.fromMap(newMap)
  }))
```